src/coreclr/gc/unix/gcenv.unix.cpp

```diff
@@ -466,7 +469,7 @@ uint64_t GCToOSInterface::GetCurrentThreadIdForLogging()
 {
 #if defined(__linux__)
     return (uint64_t)syscall(SYS_gettid);
-#elif HAVE_PTHREAD_GETTHREADID_NP
+#elif HAVE_PTHREAD_GETTHREADID_NP || defined(__FreeBSD__)
```

Comment:
Wouldn't it be cleaner to add the FreeBSD check to the `HAVE_PTHREAD_GETTHREADID_NP`-check?

Reply:
This is currently the check being done by `src/coreclr/gc/unix/configure.cmake`:

![image](https://user-images.githubusercontent.com/411338/134003694-114f6b0e-6265-46d1-8a0e-ce49eb8d3f61.png)

I guess that fails because it lacks the relevant include. Any way to shoehorn our `#include <pthread_np.h>` into this (or a similar test)? Or just create our own similar test for the BSDs? `HAVE_PTHREAD_NP_GETTHREADID_NP`?

Reply:
Yeah the check needs `#include <pthread_np.h>` but that might break the OS(s) that already pass this check.

Reply:
I'm really not a cmake-expert, so I'm not sure what the cleanest way to solve this fairly simple problem is. :(

Reply:
I don't think you can define this test for `HAVE_PTHREAD_NP_GETTHREADID_NP` more than once. I'm not a CMake guru though.

Reply:
Note, I suggested adding a new test along-side the other one (`HAVE_PTHREAD_GETTHREADID_NP` and `HAVE_PTHREAD_NP_GETTHREADID_NP`)... And then you can define a global variable based on either of those 2 passing, something like the following quasi-code:

````c-make
SET(HAVE_PTHREAD_GETTHREADID,
    HAVE_PTHREAD_GETTHREADID_NP || HAVE_PTHREAD_NP_GETTHREADID_NP`)
```` 

If you catch my drift.

Reply:
Then the test goes from being
`#elif HAVE_PTHREAD_GETTHREADID_NP || defined(__FreeBSD__)`
to
`#elif HAVE_PTHREAD_GETTHREADID_NP || defined(HAVE_PTHREAD_NP_GETTHREADID_NP)`
which seems to me to be a bit confusing - and probably less clear than just `__FreeBSD__`.
I don't know - more work needed here too 😀

Reply:
OK, maybe not the prettiest but I think it's an improvement, I've isolated the change to the cmake file. I'm still happy to take changes for better or more canonical ways of doing this.
```cmake
if(HAVE_PTHREAD_NP_H)
  set(PTHREAD_NP_H_INCLUDE "#include <pthread_np.h>")
endif()

check_cxx_source_compiles("
    #include <pthread.h>
    ${PTHREAD_NP_H_INCLUDE}
    #include <stdint.h>
```